homeassistant/components/minecraft_server/__init__.py

```diff
@@ -0,0 +1,335 @@
+
+        # Server data
+        self._name = config_data[CONF_NAME]
+        self._host = config_data[CONF_HOST]
+        self._port = config_data[CONF_PORT]
+        self._scan_interval = config_data[CONF_SCAN_INTERVAL]
+        self._server_online = False
+        self._stop_periodic_update = None
+
+        _LOGGER.debug(
+            "Initializing '%s' (host='%s', port=%s, scan_interval=%s)",
+            self._name,
+            self._host,
+            self._port,
+            self._scan_interval,
+        )
+
+        # 3rd party library instance
+        self._mcstatus = MCStatus(self._host, self._port)
+
+        # Data provided by 3rd party library
+        self._description = None
+        self._version = None
+        self._protocol_version = None
+        self._latency = None
+        self._players_online = None
+        self._players_max = None
+        self._players_list = None
+
+        # Dispatcher signal name
+        self._signal_name = SIGNAL_NAME_PREFIX + self._unique_id
+
+    @property
+    def name(self) -> str:
+        """Return server name."""
+        return self._name
+
+    @property
+    def unique_id(self) -> str:
+        """Return server unique ID."""
+        return self._unique_id
+
+    @property
+    def host(self) -> str:
+        """Return server host."""
+        return self._host
+
+    @property
+    def port(self) -> int:
+        """Return server port."""
+        return self._port
+
+    @property
+    def signal_name(self) -> str:
+        """Return dispatcher signal name."""
+        return self._signal_name
+
+    @property
+    def description(self) -> str:
+        """Return server description."""
+        return self._description
+
+    @property
+    def version(self) -> str:
+        """Return server version."""
+        return self._version
+
+    @property
+    def protocol_version(self) -> int:
+        """Return server protocol version."""
+        return self._protocol_version
+
+    @property
+    def latency_time(self) -> int:
+        """Return server latency time."""
+        return self._latency
+
+    @property
+    def players_online(self) -> int:
+        """Return online players on server."""
+        return self._players_online
+
+    @property
+    def players_max(self) -> int:
+        """Return maximum number of players on server."""
+        return self._players_max
+
+    @property
+    def players_list(self) -> List[str]:
+        """Return players list on server."""
+        return self._players_list
+
+    @property
+    def online(self) -> bool:
+        """Return server connection status."""
+        return self._server_online
+
+    def stop_periodic_update(self) -> None:
+        """Stop periodic execution of update method."""
+        if self._stop_periodic_update is not None:
+            self._stop_periodic_update()
+        else:
+            _LOGGER.debug(
+                "Listener was not started, stopping of periodic update skipped"
```

Comment:
When would this happen?

Reply:
When `async_update()` of class `MinecraftServer` was never called, like in the config flow. There I just create an instance to ping the server in order to check host and port. `async_update()` is not called there.

Reply:
If we don't call it, it won't be a problem, right?